src/scanpy/preprocessing/_pca/_dask_sparse.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, cast, overload
+
+import numpy as np
+import scipy.linalg
+from numpy.typing import NDArray
+
+from .._utils import _get_mean_var
+
+if TYPE_CHECKING:
+    from typing import Literal
+
+    from numpy.typing import DTypeLike
+    from scipy import sparse
+
+    from ..._compat import DaskArray
+
+    CSMatrix = sparse.csr_matrix | sparse.csc_matrix
+
+
+@dataclass
+class PCASparseDask:
+    n_components: int | None = None
+
+    def fit(self, x: DaskArray) -> PCASparseFit:
+        # this method makes `self` into the fitted version
+        self.__class__ = PCASparseFit
+        self = cast(PCASparseFit, self)
+
+        assert isinstance(x.shape, tuple)
+        self.n_components_ = (
+            min(x.shape) if self.n_components is None else self.n_components
+        )
+        self.n_samples_ = x.shape[0]
+        self.n_features_in_ = x.shape[1] if x.ndim == 2 else 1
```

Comment:
Maybe make a little cleaner now that we are planning on allowing `X` to have more than one dimension.  Should we raise a `ValueError` here?

Reply:
good point!